### ISSUE TITLE
Opera crashes when saving an image

### ISSUE THREAD

**Issue by coretemp** [NixOS/nixpkgs#47968]
## Issue description
Go to http://www.toledorocket.com/perftest/gfx/test.jpg
Press Control-s
Instant crash

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.73, NixOS, 18.09.git.7f70ebf (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```
opera --version
53.0.2907.99
```

**Comment by coretemp** [comment]
It also crashed when uploading an image, it was just that this was the shortest repro. 

```
Fontconfig warning: "/etc/fonts/fonts.conf", line 146: blank doesn't take any effect anymore. please remove it from your fonts.conf
[28269:28269:1007/090331.206128:ERROR:sandbox_linux.cc(379)] InitializeSandbox() called with multiple threads in process gpu-process.

(opera:28240): GLib-GIO-ERROR **: 09:03:37.076: No GSettings schemas are installed on the system
Trace/breakpoint trap
```

I use Plasma5.

**Comment by coretemp** [comment]
```
 traps: opera[28240] trap int3 ip:7fc1cf3e48f1 sp:7ffc5e407390 error:0 in libglib-2.0.so.0.5600.0[7fc1cf390000+116000]
```

**Comment by jtojnar** [comment]
Yup, you need to add `wrapGAppsHook` to Opera’s `nativeBuildInputs`. In the future, we may hard-code the schemas paths to libraries, which would allow us to drop the wrapper, but for now, it is necessary.

**Comment by coretemp** [comment]
@jtojnar I did what you said, but it just said that it was "not an ELF executable". 

Opera currently has not nativeBuildInputs. I added an entry 

`nativeBuildInputs = [ wrapGAppsHook]`

and added `, wrapGAppsHook` to e.g. line 5 of that file (something in the parameter list). 

Can you try yourself? Or propose something that does work?

**Comment by jtojnar** [comment]
Adding `wrapGAppsHook` should be enough to get rid of “No GSettings schemas are installed on the system” error. Are you sure the error is the same after adding it and rebuilding?

**Comment by coretemp** [comment]
@jtojnar It does *not* build with what you described.

**Comment by coretemp** [comment]
@jtojnar I have a PR open for a bump to 56.X to show that I wanted to fix it. I think it would be best if you work based on top of that to try to make it work.

**Comment by coretemp** [comment]
@jtojnar I have added you to my repo, so you can do whatever you want to the feature/opera branch. Not sure whether you would use it or not.

**Comment by jtojnar** [comment]
I am not near my computer at the moment. What error does the build fail with? One thing I see that is wrong with the expression is that the `installPhase` is overridden without running the pre and post hooks.
